src/log/aol.rs

```diff
@@ -183,65 +187,49 @@ impl Aol {
             )));
         }
 
-        let mut r = 0;
-
-        // Read data from the appropriate segment
-        match self.read_segment_data(&mut buf[r..], segment_id, read_offset) {
-            Ok(bytes_read) => {
-                r += bytes_read;
-            }
-            Err(e) => match e {
-                Error::Eof => {
-                    return Err(Error::Eof);
-                }
-                _ => return Err(e),
-            },
+        // Get segment from cache (works for both active and inactive segments)
+        let segment = self.segment_cache.get_or_insert_with(&segment_id, || {
```

Comment:
As a follow-up, maybe explore the possibility of keeping the active segment out of the cache for reads.

Reply:
I just wanted to avoid having an RWLock around the active segment. My thought is that since write_all writes to the os page cache in the kernel space, if a new file descriptor is opened to the same active segment, it can still read the data as we are not buffering anything in user space. 

Do you think there could be any issues with this?